quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/model/values/ValuesRule.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.quarkus.model.values;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.redhat.quarkus.commons.EnumItem;
+import com.redhat.quarkus.commons.ExtendedConfigDescriptionBuildItem;
+import com.redhat.quarkus.model.PropertiesModel;
+
+/**
+ * Values rule to manage values for some property. A values rule is composed by
+ * 
+ * <ul>
+ * <li>a matcher to know if a given property match the values rule</li>
+ * <li>the list of values {@link EnumItem}.</li>
+ * </ul>
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class ValuesRule {
+
+	private PropertyMatcher matcher;
+
+	private List<EnumItem> values;
+
+	private transient boolean valuesCleaned;
+
+	public PropertyMatcher getMatcher() {
+		return matcher;
+	}
+
+	public void setMatcher(PropertyMatcher matcher) {
+		this.matcher = matcher;
+	}
+
+	public void setValues(List<EnumItem> values) {
+		this.values = values;
+	}
+
+	public List<EnumItem> getValues() {
+		cleanValues();
+		return values;
+	}
+
+	private void cleanValues() {

```

Comment:
should be done only once, in setValues(), might also want to make values immutable after cleaning everything

Reply:
I'm using GSON to populate this values and it uses field, in other words setValues is never called.